com.sap.adt.abapcleaner/src/com/sap/adt/abapcleaner/programbase/CommandLineArgs.java

```diff
@@ -128,11 +155,50 @@ public static CommandLineArgs create(Persistency persistency, String[] args) {
 				++i;
 		}
 		
-		if (!overwrite && !StringUtil.isNullOrEmpty(targetPath) && persistency.fileExists(targetPath)) {
-			errors.append("Target file already exists; please use " + OPT_OVERWRITE + " to allow overwriting: " + targetPath).append(LINE_SEP);
+		if (sourceDir != null && sourceCode != null) {
+			errors.append("Source was supplied multiple times; please use " + OPT_SOURCE_FILE + " or " + OPT_SOURCE_CODE + " or " + OPT_SOURCE_DIR + " only once.").append(LINE_SEP);
+		}
+		
+		if (!StringUtil.isNullOrEmpty(sourceDir)) {
+			if (StringUtil.isNullOrEmpty(targetDir)) {
+				errors.append("Incomplete option: " + OPT_TARGET_DIR + " is required if the option " + OPT_SOURCE_DIR + " is provided").append(LINE_SEP);
+			} else {
+				if (cleanupRange != null)
```

Comment:
These new checks and error messages are great! However, these new branches are currently not covered by the tests in CommandLineArgsTest (and we're very proud of our test coverage :-). If you like to try, please do, otherwise, I could add them. 

Not sure whether you can execute the tests from your IDE?  (menu "Run -> Run Configurations…" -> new JUnit configuration, then in tab "Test", select "Run all tests in the selected project …: com.sap.adt.abapcleaner.test". Sorry, creating a documentation for contributors is still on my list…

As you can see in the CommandLineArgsTest.setup() method, there is a PersistencyDouble with some helper methods to prepare "fake" directories (.prepareDirectory()) and files (.prepareFile()), which are NOT stored on the local computer during test execution, but only simulated inside the PersistencyDouble. With that, you can easily trigger errors due to missing / existing directories and files (thus reaching the various branches), without actually writing something on the hard disk.